### PR TITLE
feat(server): add pidfile parameter.

### LINKED ISSUE #209: --pidfile parameter for the dragonflydb binary
`--pidfile=</path/to/dragonflybd.pid>` would make it easier to integrate it with [monit](https://mmonit.com/monit/). When this param is specified, dragonflydb would write its pid into specified file or fail if it's possible e.g. due to permission.

### PR DESCRIPTION
**Pull request by iko1**
Fixes #209
Signed-off-by: iko1 <me@remotecpp.dev>

### CODE REVIEW

File: src/server/dfly_main.cc [hunk 1]
```diff
@@ -199,6 +217,13 @@ Usage: dragonfly [FLAGS]
   CHECK_LT(kver.major, 99u);
   dfly::kernel_version = kver.kernel * 100 + kver.major;
 
+  string pidfile_path = GetFlag(FLAGS_pidfile);
```
Comment (romange) [inline src/server/dfly_main.cc:220]: the feature is a bit more complicated than that. You also need to remove the file upon exit.
see https://github.com/redis/redis/blob/d00b8af89265c501fb4a0cdd546702b90432a896/src/server.c#L421

for example

File: src/server/dfly_main.cc [hunk 2]
```diff
@@ -132,6 +134,22 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
...
 
+bool CreatePidFile(const string& path) {
+    ofstream ofs;
+    ofs.exceptions(ios::failbit | ios::badbit);
+
+    try {
+      ofs.open(path, ios::trunc);
+      ofs << getpid();
+      ofs.close();
+    } catch (ios::failure& ex) {
```
Comment (romange) [inline src/server/dfly_main.cc:145]: I also do not use c++ exceptions. These are poorly designed in c++ and you can see that I always return errors explicitly.

**Review by romange — CHANGES_REQUESTED** [review]

File: src/server/dfly_main.cc (hunk 1 above)
Comment (iko1) [reply src/server/dfly_main.cc:228]: Can you elaborate on your comment here? I'm unsure where to place the file deletion in dfly_main.cc and how to ensure the deletion function will be called at the server exit.

Comment (romange) [reply src/server/dfly_main.cc:228]: what you did at lines 265-267 is good. Was it always there? I did not see it before.

**Review by romange — APPROVED** [review]

File: src/server/dfly_main.cc (hunk 1 above)
Comment (iko1) [reply src/server/dfly_main.cc:228]: It just added in the latest commit after you asked for the removal of the file upon exit.

**Pull request merged**